### PR TITLE
Add Model Evaluation assets to azureml-assets

### PR DESCRIPTION
**Pull request by ankushbhatia2**
Adding Model Prediction, Compute Metrics, Evaluate Model and Model Evaluation Pipeline components and model evaluation Custom Environment from azureml-asset.

### CODE REVIEW

File: training/model_evaluation/components/evaluate_model/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: evaluate_model
+display_name: Evaluate Model
+description: Evaluate MLFlow models for supported task types.
+
+version: 0.0.1.preview
+
+tags:
```
Comment (ShipraJain01) [inline training/model_evaluation/components/evaluate_model/spec.yaml:8]: Add Preview:""  tag

**Comment by ankushbhatia2** [comment]
This PR depends on #479

**Review by vizhur — CHANGES_REQUESTED** [review]
please split this PR to docs, envs and components that depend on it

File: training/model_evaluation/components/compute_metrics/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    type: uri_file 
+    optional: true
+    description: "Additional parameters required for evaluation."
+
+outputs:
+  evaluation_result:
+    type: uri_folder
+
+code: ../../src
+environment: azureml:model-evaluation:0.0.1-preview
```
Comment (vizhur) [inline training/model_evaluation/components/compute_metrics/spec.yaml:70]: this should have integer version

Comment (ankushbhatia2) [reply training/model_evaluation/components/compute_metrics/spec.yaml:70]: We have kept the version  as 0.0.1-preview in Env asset. https://github.com/Azure/azureml-assets/blob/c1b563e1b3799ac5ca6215ebf5582c96b71f7618/training/model_evaluation/environment/asset.yaml#L2

Comment (ankushbhatia2) [reply training/model_evaluation/components/compute_metrics/spec.yaml:70]: Shall I make it to auto?

Comment (elesel) [reply training/model_evaluation/components/compute_metrics/spec.yaml:70]: version: auto is set in the asset.yaml file: https://github.com/Azure/azureml-assets/wiki/Environments#assetyaml

Comment (ankushbhatia2) [reply training/model_evaluation/components/compute_metrics/spec.yaml]: Updated

**Review by elesel — DISMISSED** [review]

**Review by ShipraJain01 — DISMISSED** [review]

**Review by vizhur — APPROVED** [review]

**Review by ShipraJain01 — APPROVED** [review]

**Review by ShipraJain01 — APPROVED** [review]

File: training/model_evaluation/components/compute_metrics/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    type: uri_file 
+    optional: true
+    description: "Additional parameters required for evaluation."
+
+outputs:
+  evaluation_result:
+    type: uri_folder
+
+code: ../../src
+environment: azureml:model-evaluation:1
```
Comment (ShipraJain01) [inline training/model_evaluation/components/compute_metrics/spec.yaml]: please map latest environment

**Review by ShipraJain01 — APPROVED** [review]

**Pull request merged**